client/repo.c

```diff
@@ -983,6 +978,10 @@ TDNFGetRepoMD(
     *ppRepoMD = pRepoMD;
 
 cleanup:
+    if (!IsNullOrEmptyString(pszTmpRepoDataDir))
+    {
+        TDNFRemoveTmpRepodata(pszTmpRepoDataDir);
```

Comment:
I see TDNFRemoveTmpRepodata is only used here and we are not using the error code returned.Shall we modify TDNFRemoveTmpRepodata function?

Reply:
We can add a return error check for TDNFRemoveTmpRepodata along with dwError == ERROR_TDNF_CHECKSUM_VALIDATION_FAILED and add a pr_crit print saying "Downloaded repoMD file shasum mismatch, please remove the tmp directory path". This will alert the user to remove malicious repomd files manually, if tmp directory removal is failed from tdnf.

Reply:
Now errno is getting set while nftw is getting failed so should we check dwError is not 0 and print saying above or instead if dwError = errno replaced with ERROR_TDNF_CHECKSUM_VALIDATION_FAILED ?

Reply:
It should be something like below:
```
if (!IsNullOrEmptyString(pszTmpRepoDataDir))
{
        if (TDNFRemoveTmpRepodata(pszTmpRepoDataDir) && dwError == ERROR_TDNF_CHECKSUM_VALIDATION_FAILED )
        {
                pr_crit("Downloaded repomd shasum mismatch, failed to delete %s file.  please delete.", pszTmpRepoDataDir)
         }
}
```

Reply:
done